core/src/sumeragi/main_loop.rs

```diff
@@ -404,6 +406,13 @@ impl Sumeragi {
     ) -> Option<VotingBlock<'state>> {
         let mut state_block = state.block();
 
+        if state_block.height() == 1 && block.header().height == 1 {
```

Comment:
I believe this is already checked in `ValidBlock::validate`

Reply:
Yes but `ValidBlock::validate` will print warning about it, so I added this check to avoid the warning

Reply:
ok, but you can inline this check into `ValidBlock::validate` and skip writing warning in there then

Reply:
it's fine